app/models/person.rb

```diff
@@ -74,4 +86,91 @@ def full_name(format = :default)
       raise ArgumentError, "Unsupported format: #{format}"
     end
   end
+
+  def add_role_and_group(*names)
+    transaction do
+      names.each do |name|
+        user.add_role(name, Current.organization)
+      end
+      add_group(*names)
+    end
+  end
+
+  def add_group(*names)
+    names.map(&:to_s).uniq.each do |name|
+      next unless Group.names.key?(name)
+      group = Group.find_or_create_by!(name: name)
```

Comment:
You'll have to remind me, if I call this within the Org scoped controllers where we set a tenant context, this will add the `org_id` to the `group` record? But, if I call it outside of that context e.g., in the console, it won't have an org scope and won't know what `org_id` to use, right? So, do we need to handle that ability?
`Group.find_or_create_by!(name: name)`

I am looking at this late on Saturday night, so I could be derping here.

Reply:
Yeah I think I am just reminding myself of a fundamental approach of using ActsAsTenant...it has been a while.

Reply:
I agree. I like the idea explicit validation here rather than relying on and gems. Devs could make mistakes and in this case it could be very costly eg making someone a staff in a different organization.